internal/pkg/build/files/files.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package files
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func expandPath(path string) ([]string, error) {
+	var output, stderr bytes.Buffer
+	cmd := exec.Command("/bin/sh", "-c", "echo "+path)
+	cmd.Stdout = &output
+	cmd.Stderr = &stderr
+	if err := cmd.Run(); err != nil {
+		return nil, fmt.Errorf("%s: %s", err, stderr.String())
+	}
+
+	return strings.Fields(output.String()), nil
+}
+
+// Join wraps filepath.Join() and appends a trailing '/' if originally specified
+// in the last element since that has meaning to the cp command
```

Comment:
but then this is not a generic join function, but a special-purpose one. Should this be called Join?

Reply:
Is a `Join` function something that _should_ clean the result in a particular way? If so, then I agree the name should be changed, but if not I feel the comment is enough to explain the particular behavior of this function since in all cases except those involving a trailing slash will behave the same way

Reply:
I'm thinking about the person reading the code: it's easy to see files.Join(...) and mistake it for filepath.Join(...). Also thinking of the case of someone writing new code and seeing that "file.Join" is used. They would think that for whatever reason, in this code that's how you join paths (instead of using filepath.Join).

I'm trying to think of a name for this operation and I cannot think of something descriptive enough. Basically when you ask it to join "/a" and "b/" it returns "/a/b/", but if you ask it to join "/a" and "b" it returns "/a/b". It's not quite the reverse of "dirname" because dirname removes the last element of a path, no matter what that element might be.

files.AddPrefix? And maybe change to `func AddPrefix(prefix, path string) string` (since all the uses use only two arguments)?

Reply:
Makes more sense to me, will update